File: catalog/backends/python-django/patterns/cpu_intensive_service.py
from django.http import JsonResponse

# Pattern: Profiling — CPU-intensive nested loop
# Adapt: replace with domain-relevant aggregation logic


def aggregate(request):
    counts = {}
    for _ in range(100):
        counts.clear()
        for j in range(10000):
            key = f"bucket-{j % 50}"
            counts[key] = counts.get(key, 0) + 1
    return JsonResponse({"buckets": len(counts), "iterations": 100 * 10000})

File: catalog/backends/php-laravel/patterns/CpuIntensiveService.php
<?php

use Illuminate\Http\JsonResponse;

// Pattern: Profiling — CPU-intensive nested loop
// Adapt: replace with domain-relevant aggregation logic

function aggregate(): JsonResponse
{
    $counts = [];
    for ($i = 0; $i < 100; $i++) {
        $counts = [];
        for ($j = 0; $j < 10000; $j++) {
            $key = 'bucket-' . ($j % 50);
            $counts[$key] = ($counts[$key] ?? 0) + 1;
        }
    }
    return response()->json(['buckets' => count($counts), 'iterations' => 100 * 10000]);
}
